### ISSUE TITLE
IPN Sent Successfully not working

### ISSUE THREAD

**Issue by casha1995** [OrahKokos/coinpayments#44]
after upgrading to the new version , and following the steps , i get ipn , with no problems , but in my coin-payment account , sent successfully appears as no (num tries left) , even when i get ipn calls to the back-end receiving it successfully with no errors or no problems , but this didn't happen in older version , so can some one help in this matter 
here is the code that i'm trying after update to version 2.x.x

```
const { verify } = require(`coinpayments-ipn`);
const CoinpaymentsIPNError = require(`coinpayments-ipn/lib/error`);

const {
  MERCHANT_ID,
  IPN_SECRET
} = process.env;

app.post('/notify', function (req, res, next) {
  if(!req.get(`HMAC`) || !req.body || !req.body.ipn_mode || req.body.ipn_mode !== `hmac` || MERCHANT_ID !== req.body.merchant) {
    return next(new Error(`Invalid request`));
  }

  let isValid, error;

  try {
    isValid = verify(req.get(`HMAC`), IPN_SECRET, req.body);
  } catch (e) {
    error = e;
  }
  
  if (error && error instanceof CoinpaymentsIPNError) {
    return next(error);
  }
  
  if (!isValid) {
    return next(new Error(`Hmac calculation does not match`));
  }

  return next();
}, function (req, res, next) {
  console.log(`Process payment notification`);
  console.log(req.body);
  // then some operations in db
  return next();
});
```

**Comment by OrahKokos** [comment]
@casha1995 I'm not sure I follow what is the issue.
Can you rephrase your question?
Also please open this issue on the correct repo: https://github.com/OrahKokos/coinpayments-ipn
